tasks_week3_omer_gertler1/src/Zoo.cpp

```diff
@@ -0,0 +1,179 @@
+#include"Zoo.h"
+#include <string>
+using std::cout;
+using std::cin;
+using std::endl;
+ 
+
+Zoo::Zoo()
+{
+	Location loc;
+	loc.setRow(rand() % ROWS);
+	loc.setCol(rand() % COLUMNS);
+	Lion* newLion = new Lion("Simba", loc);
```

Comment:
bad allocation, rhis will lead to memory leak, you can use it this way animals.push_back(new Monkey("Rafiki", loc));
also you can use the create function and pass the type and name, it's more generic and effeicent